src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/WindowsRuntime/WindowsRuntimeMarshalSupport.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Internal.Runtime.InteropServices.WindowsRuntime
+{
+    public static class WindowsRuntimeMarshalSupport
+    {
+        /// <summary>
+        /// Get EventHandler for specified EventRegistrationToken if it exists.
+        /// An internal contract between System.Private.Corelib and System.Runtime.WindowsRuntime
+        /// </summary>
+        public static T GetEventHandlerFromEventRegistrationToken<T>(EventRegistrationTokenTable<T> table, EventRegistrationToken token)  where T : class
```

Comment:
This also removes the token from the table. It is pretty non-obvious from the name.

Reply:
I think we should just add this as public API to EventRegistrationTokenTable directly:

`bool RemoveEventHandler(EventRegistrationToken token, out T handler)`

There are other non-exposed public APIs on EventRegistrationTokenTable, so having one more is not a big deal.

Reply:
Added RemoveEventHandler in EventRegistrationTokenTable